solidity/contracts/Oracle.sol

```diff
@@ -77,6 +78,7 @@ contract Oracle is Ownable {
     hasInternalId(_internalId)
   {
     Callback memory callback = callbacks[_internalId];
+    // solium-disable-next-line security/no-low-level-calls
```

Comment:
Could use an explanation here.

Reply:
It shows a warning for any `call` method and below we're using

```
callback.addr.call(callback.functionId, ...)
```

Hence the `no-low-level-calls`. We need `call` to work with `functionId`. Are you looking for a comment like the one below?

```
// callback.addr.call(callback.functionId, ...) is needed to invoke the functionId
```

Let me know what you're looking for. A suggestion would go a long way.

Reply:
This is a pretty unusual mechanism which I'll admit to not entirely understanding right now, so not sure my suggestion is going to help all that much. Nevertheless `call` is advised against presumably because it can allow for arbitrary code execution. What have we done to ensure that this is not the case? What safeguards do we have against abuse? Would be nice to have that documented somewhere.

Reply:
Well, we actually want arbitrary code execution.

```solidity
  struct Callback {
    bytes32 externalId;
    uint256 amount;
    address addr;  // safe guard that only function ids are called on this addr
    bytes4 functionId; // what's sent to `call`
  }
```

```solidity
    ChainlinkLib.Run memory run = newRun(jobId, this, "fulfill(bytes32,bytes32)");
```

Functions aren't first class parameters, so the next best thing is the function id, which we generate off its signature.

The safe guard we put in place to prevent abuse is to only call functions on the requester's address. I'll document that.

Reply:
Yeah, I think what we don't want is "unexpected arbitrary code execution" i.e. for a user to be able to somehow hijack `functionId` and actually call their function to transfer all of our LINK to address 0.